packages/ds-ext/src/app/App.tsx

```diff
@@ -5,58 +5,13 @@ import { VsCodeToast } from './VsCodeToast';
 import { onDrop } from './onDrop';
 import { createVsCodeClient } from './createVsCodeClient';
 
-export const fileUri = window.initialData.fileUri;
-
 export default function App() {
-  const [diagram, setDiagram] = useState<Diagram | undefined>();
-
-  useEffect(() => {
-    if (window.vscode) {
-      window.vscode.postMessage({
-        type: 'getDirtyFileContent',
-      });
-    }
-
-    const handleMessage = (event: any) => {
-      const message = event.data;
-
-      if (message.type === 'dirtyFileContent') {
-        setDiagram(
-          message.fileContent
-            ? new Diagram(JSON.parse(message.fileContent))
-            : new Diagram()
-        );
-      }
-    };
-
-    window.addEventListener('message', handleMessage);
-
-    // Cleanup the listener on unmount
-    return () => {
-      window.removeEventListener('message', handleMessage);
-    };
-  }, []);
-
+  const client = createVsCodeClient(window.vscode);
   const handleChange = useCallback(
     debounce(async (diagram: Diagram) => {
-      // Construct the message payload with updated diagram data
-      const updatedData = {
-        type: 'updateDiagram',
-        fileUri, // Include the file URI to specify the target file
-        diagram: JSON.stringify(diagram),
-      };
-
-      // Send the message to VS Code extension
-      window.vscode.postMessage(updatedData);
+      client?.updateDiagram?.(diagram);
```

Comment:
Do we need the safe(?) operators here? At this time, could/should we enforce a client always to be present?